### PR TITLE
Add devenum

### PR DESCRIPTION
**Pull request by Zemogiter**
I have a problem. When the verb is attempting to extract from the .exe file it crashes with this:
```
javax.script.ScriptException: ReferenceError: "AppResource" is not defined in <eval> at line number 29
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(NashornScriptEngine.java:477)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:461)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:413)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:409)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:162)
	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
	at org.phoenicis.scripts/org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: <eval>:29 ReferenceError: "AppResource" is not defined
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.error(ECMAErrors.java:57)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.referenceError(ECMAErrors.java:319)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.referenceError(ECMAErrors.java:291)
	at jdk.scripting.nashorn/jdk.nashorn.internal.objects.Global.__noSuchProperty__(Global.java:1616)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$10087$915AA$\^eval\_.run#L:24(<eval>:29)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$10007$553$\^eval\_$cu1$restOf.go([engines, wine, quick_script, installer_script]:75)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$9975$592$\^eval\_.run(<eval>:16)
	at jdk.nashorn.javaadapters.org_phoenicis_scripts_Installer.run(Unknown Source)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$9974$\^eval\_$cu1$restOf.:program(<eval>:59)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:655)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:513)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:527)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:456)
	... 10 more
```

@qparis were you getting similiar error with paint?

### CODE REVIEW

**Comment by plata** [comment]
See https://github.com/PhoenicisOrg/phoenicis/issues/1600

**Comment by Zemogiter** [comment]
It seems this depends on the package you want to extract cause the June2010 directX exe file that is used in `d3dx10` verb is extracted without error. Unfortunately that file does not contain the devenum.dll
I'm considering looking for the exact same file online.

**Comment by qparis** [comment]
> I have a problem. When the verb is attempting to extract from the .exe file it crashes with this:
> 
> ```
> javax.script.ScriptException: ReferenceError: "AppResource" is not defined in <eval> at line number 29
> 	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(NashornScriptEngine.java:477)
> 	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:461)
> 	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:413)
> 	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:409)
> 	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.eval(NashornScriptEngine.java:162)
> 	at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
> 	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornEngine.eval(NashornEngine.java:50)
> 	at org.phoenicis.scripts/org.phoenicis.scripts.nashorn.NashornScriptInterpreter.runScript(NashornScriptInterpreter.java:35)
> 	at org.phoenicis.scripts/org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$runScript$0(BackgroundScriptInterpreter.java:35)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> 	at java.base/java.lang.Thread.run(Thread.java:834)
> Caused by: <eval>:29 ReferenceError: "AppResource" is not defined
> 	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.error(ECMAErrors.java:57)
> 	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.referenceError(ECMAErrors.java:319)
> 	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.referenceError(ECMAErrors.java:291)
> 	at jdk.scripting.nashorn/jdk.nashorn.internal.objects.Global.__noSuchProperty__(Global.java:1616)
> 	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$10087$915AA$\^eval\_.run#L:24(<eval>:29)
> 	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$10007$553$\^eval\_$cu1$restOf.go([engines, wine, quick_script, installer_script]:75)
> 	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$9975$592$\^eval\_.run(<eval>:16)
> 	at jdk.nashorn.javaadapters.org_phoenicis_scripts_Installer.run(Unknown Source)
> 	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$9974$\^eval\_$cu1$restOf.:program(<eval>:59)
> 	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:655)
> 	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:513)
> 	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:527)
> 	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:456)
> 	... 10 more
> ```
> 
> @qparis were you getting similiar error with paint?

Not the same problem apparently

**Comment by Zemogiter** [comment]
I've retested the Lego Rock Raiders script with the new verbs commented out and the error is still there. I was wrong the entire time. It's this line:
```
var registrySettings = new AppResource().application([TYPE_ID, CATEGORY_ID, APPLICATION_ID]).get("fix.reg");
wine.regedit().patch(registrySettings);
```
that makes this error. So the extraction was working this entire time smh

**Comment by Zemogiter** [comment]
Just tested it in the Lego Rock Raiders script. This verb is working and ready for review.

File: Engines/Wine/Verbs/devenum/script.js [hunk 1]
```diff
@@ -33,3 +33,20 @@ Wine.prototype.devenum = function () {
 
     return this;
 };
+
+/**
+ * Verb to install all the necessary things to run winevulkan (even inside wine mainline or newest wine-staging)
```
Comment (plata) [inline Engines/Wine/Verbs/devenum/script.js]: copy paste error

File: Engines/Wine/Verbs/devenum/script.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    this.regsvr32().install("devenum.dll");
+    this.overrideDLL()
+        .set("native", ["devenum"])
+        .do();
+
+    return this;
+};
+
+/**
+ * Verb to install all the necessary things to run devenum (even inside wine mainline or newest wine-staging)
```
Comment (plata) [inline Engines/Wine/Verbs/devenum/script.js]:
```suggestion
  * Verb to install devenum
```

**Comment by Zemogiter** [comment]
Ready to be merged.

**Pull request merged**